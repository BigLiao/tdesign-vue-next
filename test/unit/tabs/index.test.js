import { mount } from '@vue/test-utils';
import Tabs from '@/src/tabs/index.ts';

// every component needs four parts: props/events/slots/functions.
describe('Tabs', () => {
  // test props api
  describe(':props', () => {
    it('', () => {
      const wrapper = mount({
        render() {
          return <Tabs />;
        },
      });
      expect(wrapper.isEmpty()).toBe(false);
    });
    it(':theme', () => {
      const wrapper = mount({
        render() {
          return <Tabs theme={'card'} />;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':activeName', () => {
      const wrapper = mount({
        render() {
          return <Tabs activeName={1} />;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':size', () => {
      const wrapper = mount({
        render() {
          return <Tabs activeName={'large'} />;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':tabPosition', () => {
      const wrapper = mount({
        render() {
          return <Tabs tabPosition={'left'} />;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':addable', () => {
      const wrapper = mount({
        render() {
          return <Tabs addable={true} />;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  // test events
  describe('@event', () => {
    it('@add', async () => {
      const wrapper = mount({
        render() {
          return <Tabs theme={'card'} addable={true} />;
        },
      });
      const tabs = wrapper.find(Tabs);
      await tabs.find('.t-tabs__add-btn').trigger('click');
      expect(tabs.emitted().add).toBeTruthy();
    });
  });

  // // test slots
  // describe('<slot>', () => {
  //   it('', () => null);
  // });

  // // test exposure function
  // describe('function', () => {
  //   it('', () => null);
  // });
});
