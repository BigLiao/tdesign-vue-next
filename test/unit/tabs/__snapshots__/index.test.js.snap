// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs :props :activeName 1`] = `
<div class="t-tabs">
  <div class="t-tabs__header t-is-top">
    <div class="t-tabs__nav">
      <!---->
      <div class="t-tabs__nav-container t-is-top">
        <!---->
        <!---->
        <div class="t-tabs__nav-scroll">
          <div class="t-tabs__nav-wrap" style="transform: translateX(0px);">
            <div class="t-tabs__bar t-is-top"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="t-tabs__content"></div>
</div>
`;

exports[`Tabs :props :addable 1`] = `
<div class="t-tabs">
  <div class="t-tabs__header t-is-top">
    <div class="t-tabs__nav">
      <!---->
      <div class="t-tabs__nav-container t-is-top">
        <!---->
        <!---->
        <div class="t-tabs__nav-scroll">
          <div class="t-tabs__nav-wrap" style="transform: translateX(0px);">
            <div class="t-tabs__bar t-is-top"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="t-tabs__content"></div>
</div>
`;

exports[`Tabs :props :size 1`] = `
<div class="t-tabs">
  <div class="t-tabs__header t-is-top">
    <div class="t-tabs__nav">
      <!---->
      <div class="t-tabs__nav-container t-is-top">
        <!---->
        <!---->
        <div class="t-tabs__nav-scroll">
          <div class="t-tabs__nav-wrap" style="transform: translateX(0px);">
            <div class="t-tabs__bar t-is-top"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="t-tabs__content"></div>
</div>
`;

exports[`Tabs :props :tabPosition 1`] = `
<div class="t-tabs">
  <div class="t-tabs__header t-is-left">
    <div class="t-tabs__nav">
      <!---->
      <div class="t-tabs__nav-container t-is-left">
        <!---->
        <!---->
        <div class="t-tabs__nav-scroll">
          <div class="t-tabs__nav-wrap" style="transform: translateX(0px);">
            <div class="t-tabs__bar t-is-left"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="t-tabs__content"></div>
</div>
`;

exports[`Tabs :props :theme 1`] = `
<div class="t-tabs">
  <div class="t-tabs__header t-is-top">
    <div class="t-tabs__nav">
      <!---->
      <div class="t-tabs__nav-container t-tabs__nav--card t-is-top">
        <!---->
        <!---->
        <div class="t-tabs__nav-scroll">
          <div class="t-tabs__nav-wrap" style="transform: translateX(0px);">
            <!---->
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="t-tabs__content"></div>
</div>
`;
